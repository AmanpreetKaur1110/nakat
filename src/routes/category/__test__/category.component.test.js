import { render, screen } from '@testing-library/react';
import { RouterProvider, createMemoryRouter } from 'react-router-dom';
import { CategoriesContext } from '../../../contexts/categories.context';
import Category from '../category.component';

const mockCategoriesMap = {
  'oil-painting': [
    { id: 1, name: 'road-side', price: 1000 },
    { id: 2, name: 'birds', price: 1800 }
  ],
  Sketches: [
    { id: 3, name: 'sketch', price: 1500 }
  ]
};

const renderWithRouterAndContext = (category = 'oil-painting') => {
  const router = createMemoryRouter(
    [
      {
        path: '/shop/:category',
        element: <Category />,
      },
    ],
    {
      initialEntries: [`/shop/${category}`],
      future: {
        v7_relativeSplatPath: true,
        v7_startTransition: true, 
      },
    }
  );

  return render(
    <CategoriesContext.Provider value={{ categoriesMap: mockCategoriesMap }}>
      <RouterProvider router={router} />
    </CategoriesContext.Provider>
  );
};

test('renders correct category title and product names', () => {
  renderWithRouterAndContext('oil-painting');
  expect(screen.getByText('OIL-PAINTING')).toBeInTheDocument();
  expect(screen.getByText(/road-side/i)).toBeInTheDocument();
  expect(screen.getByText(/birds/i)).toBeInTheDocument();
});

test('matches the snapshot for the category component', () => {
  const { asFragment } = renderWithRouterAndContext('oil-painting');
  expect(asFragment()).toMatchSnapshot();
});

