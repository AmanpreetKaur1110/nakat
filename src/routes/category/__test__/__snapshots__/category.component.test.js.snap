// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot for the category component 1`] = `
<DocumentFragment>
  <h2
    class="sc-kpOvIu hGRBOl"
  >
    OIL-PAINTING
  </h2>
  <div
    class="sc-gDpztx kRiyyP"
  >
    <div
      class="product-card-container"
    >
      <img
        alt="road-side"
      />
      <div
        class="footer"
      >
        <span
          class="name"
        >
          Name:- road-side
        </span>
        <span
          class="price"
        >
          Rs.1000
        </span>
      </div>
      <button
        class="sc-dKREkF sc-dIMoHT bpYrpT jSjKlJ"
      >
        Addtocard 
      </button>
    </div>
    <div
      class="product-card-container"
    >
      <img
        alt="birds"
      />
      <div
        class="footer"
      >
        <span
          class="name"
        >
          Name:- birds
        </span>
        <span
          class="price"
        >
          Rs.1800
        </span>
      </div>
      <button
        class="sc-dKREkF sc-dIMoHT bpYrpT jSjKlJ"
      >
        Addtocard 
      </button>
    </div>
  </div>
</DocumentFragment>
`;
